### ISSUE TITLE
SG on Snap fails to restart after update

### ISSUE THREAD

**Issue by JackHall81** [SickGear/SickGear#1325]
#### On Linux Mint 19.3 - Kernel 4.15.0-122 - Fully up-to-date

Hello, I'm not a programmer by a long stretch so I apologize in advance for any shortcoming in posting this.

**Background:** I have recently migrated from SB to SG on my old and trusty Linux Mint 19.3 (Ubuntu-based) machine. Somehow I had managed to keep SB alive for a long time but I finally moved on to SG. Installed via snap and eventually got it up and running, even set up a systemd service to run automatically on boot
**The issue:** automatic updates in snap are "internally" working fine, but every time there is a SG update (latest from 1033 to 1038) the service fails to start. 
After digging a bit I found that the python PID that used to run in the old version is still registered and active listening on port 8081, so the new version finds a conflict on the listening port and fails to start.
Also browsing to the WebUI (localhost:8081) gives out an "traceback" error pointing to the old version directory, which may not even exist anymore.
As a workaround I have to manually do the following:
1- find the PID that is still hanging there with 'service sickgear status' 
2- stop the service 
3- kill the offending PID
4- start the service
Now SG starts happily again and all is good.
------------ | -------------
Obviously the expectation is that the update should be smooth and transparent

Maybe I'm missing a setup step? I'm pretty sure I followed the guides to the letter, and then some, as permissions needed some tweaking here and there until it started running, but then the updates issue coming later was a bit unexpected.

Thanks

**Comment by JackDandy** [comment]
Hi @JackHall81 

Thanks for raising this, I'm a full time Windows user and making the Linux snap was quite a challenge, complete up to the point where it runs inside a terminal, can be shutdown, and can be updated in that terminal without issue. 

However,  then the full time Linux chaps (you're not the first) come along and want to make it into a service, Linux chaps far more knowledgeable on all things Linux than I, could probably do this in their sleep, asking a Windows guy who created this Linux stuff over weeks of trial and error, blood sweat and tears to make it systemd. 

Where does that leave you? Well, I believe that systemd service stuff **is** your issue here.

I do not know enough of how to create one that shuts down the snap cleanly - so that the snap update process can execute as you expect. 

But... here is what I do know... 

A few years back a Debian server was loaned to me to make the SickGear software work via systemd, I managed it and one of the key things in that exercise, was that I had to create and invoke SG with param `--systemd` from within the systemd file. This systemd file is included in the SG package at `init-scripts/init.systemd` and if you look there, you will see things like `ExecStart` to run, and `restart=on-failure` and also the `wantedby` ... all of these were extra to normal things I added in order to get systemd functioning _cleanly_, that meant restarting the service, and shutting it down from `systemctl` or whatever it is called. Hopefully, this may help you in your quest, best that I can offer up to you. 

It would be something special if a fully working systemd was kindly donated from you.
